content/en/docs/started/architecture.md

```diff
@@ -4,142 +4,147 @@ description = "An overview of Kubeflow's architecture"
+  features for the offline feature store, and prepare training data for model development.
+  Usually, this step is associated with data processing tools such as Spark, Dask, Flink, or Ray.
 
-To learn more, read the following guides to the Kubeflow components:
+- In the _Model Development_ step you choose an ML framework, develop your model architecture and
+  explore the existing pre-trained models for fine-tuning like BERT or Llama.
 
-* Kubeflow includes services for spawning and managing 
-  [Jupyter notebooks](/docs/components/notebooks/). Use notebooks for interactive data 
-  science and experimenting with ML workflows.
+- In the _Model Optimization_ step you can optimize your model hyperparameters and optimize your
+  model with various AutoML algorithms such as neural architecture search and model compression.
+  During model optimization you can store ML metadata in the _Model Registry_.
 
-* [Kubeflow Pipelines](/docs/components/pipelines/) is a platform for 
-  building, deploying, and managing multi-step ML workflows based on Docker 
-  containers.
+- In the _Model Training_ step you train or fine-tune your model on the large-scale
+  compute environment. You should use a distributed training if single GPU can't handle your
+  model size. The results of the model training is the trained model artifact that you
+  can store in the _Model Registry_.
 
-* Kubeflow offers several [components](/docs/components/) that you can use
-  to build your ML training, hyperparameter tuning, and serving workloads across
-  multiple platforms.
+- In the _Model Serving_ step you serve your model artifact for online or batch inference. Your
+  model may perform predictive or generative AI tasks depending on the use-case. During the model
+  serving step you may use an online feature store to extract features. You monitor the model
+  performance, and feed the results into your previous steps in the ML lifecycle.
 
-## Example of a specific ML workflow
+### ML Lifecycle for Production and Development Phases
 
-The following diagram shows a simple example of a specific ML workflow that you
-can use to train and serve a model trained on the MNIST dataset:
+The ML lifecycle for AI applications may be conceptually split between _development_ and
+_production_ phases, this diagram explores which stages fit into each phase:
 
-<img src="/docs/images/kubeflow-gcp-e2e-tutorial-simplified.svg" 
-  alt="ML workflow for training and serving an MNIST model"
-  class="mt-3 mb-3 border border-info rounded">
+<img src="/docs/started/images/ml-lifecycle-dev-prod.drawio.svg"
+  alt="ML Lifecycle with Development and Production"
+  class="mt-3 mb-3">
+
+### Kubeflow Components in the ML Lifecycle
+
+The next diagram shows how Kubeflow components are used for each stage in the ML lifecycle:
+
+<img src="/docs/started/images/ml-lifecycle-kubeflow.drawio.svg"
+  alt="Kubeflow Components in ML Lifecycle"
+  class="mt-3 mb-3">
+
```

Comment:
We should add an introduction, and I think it might look better without the double spaces between each bullet point (but I will defer to you).

```suggestion
See the following links for more information about each Kubeflow component:

```

Reply:
I like the separation tho, given that it is easier to see different Kubeflow projects.